packages/kit/src/vite/dev/index.js

```diff
@@ -235,6 +240,16 @@ export async function dev(vite, vite_config, svelte_config) {
 					? await vite.ssrLoadModule(`/${svelte_config.kit.files.hooks}`)
 					: {};
 
+				/** @type {(env: Record<string, string>) => void} */
+				const set_env = (await import(`${runtime}/app/env/runtime.js`)).set_env;
+
+				set_env(
+					// This just makes sure no values are undefined
+					Object.entries(process.env).reduce((prev, [k, v]) => {
+						return { ...prev, [k]: v ?? '' };
+					}, {})
+				);
```

Comment:
Is that a thing that can happen? I've never seen a value in `process.env` that wasn't a `string`

```suggestion
				set_env(process.env);
```

Reply:
Looking at [`@types/node`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e296423148c71102ca1f05c6d632453c9e6e5189/types/node/process.d.ts#L112-L117)...

```ts
interface ProcessEnv extends Dict<string> {
    /**
     * Can be used to change the default timezone at runtime
     */
    TZ?: string;
}
```

...it seems that values [_can_ be undefined](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e296423148c71102ca1f05c6d632453c9e6e5189/types/node/globals.d.ts#L287-L289) as far as TypeScript is concerned:

```ts
interface Dict<T> {
    [key: string]: T | undefined;
}
```

But I've never actually seen that be the case, which makes me think maybe the types are wrong

Reply:
Oh you know what? This is type system funkiness. They _can_ be `undefined`, but only for keys which are also `undefined`. i.e. `process.env.FOO` might be undefined if `FOO` isn't an environment variable. So given that we're looping over keys which are guaranteed to exist, we can probably just hard cast it to a `Record<string, string>`.

I'll fix it if you don't. I think there's something similar in `sync` as well.